addons/crm/models/crm_lead.py

```diff
@@ -431,6 +431,20 @@ def write(self, vals):
 
         return write_result
 
+    def unlink(self):
+        # Delete those fields in order to make the Document button in the
+        # event form view invisable.
+        for lead in self:
+            events = lead.env['calendar.event'].search([
+                ('res_id', '=', lead.id),
+                ('res_model_id', '=', self._name),
+            ])
+            events.write({
+                'res_id': False,
+                'res_model_id': False,
+            })
```

Comment:
```suggestion
        events = lead.env['calendar.event'].search([
            ('res_id', '=', self.ids),
            ('res_model_id', '=', self._name),
        ])
        events.write({
            'res_id': False,
            'res_model_id': False,
        })
```

Reply:
Did you mean `self` instead of `lead` on line 437 in your suggestion?